src/Raven.Client/Documents/Commands/NextHiLoCommand.cs

```diff
@@ -14,27 +16,37 @@ public class NextHiLoCommand : RavenCommand<HiLoResult>
         private readonly DateTime _lastRangeAt;
         private readonly char _identityPartsSeparator;
         private readonly long _lastRangeMax;
+        private readonly int? _shardIndex;
 
-        public NextHiLoCommand(string tag, long lastBatchSize, DateTime lastRangeAt, char identityPartsSeparator, long lastRangeMax)
+        public NextHiLoCommand(string tag, long lastBatchSize, DateTime lastRangeAt, char identityPartsSeparator, long lastRangeMax, int? shardIndex)
         {
             _tag = tag ?? throw new ArgumentNullException(nameof(tag));
             _lastBatchSize = lastBatchSize;
             _lastRangeAt = lastRangeAt;
             _identityPartsSeparator = identityPartsSeparator;
             _lastRangeMax = lastRangeMax;
+            _shardIndex = shardIndex;
         }
 
         public override HttpRequestMessage CreateRequest(JsonOperationContext ctx, ServerNode node, out string url)
         {
-            var path = $"hilo/next?tag={Uri.EscapeDataString(_tag)}&lastBatchSize={_lastBatchSize}&lastRangeAt={_lastRangeAt:o}&identityPartsSeparator={Uri.EscapeDataString(_identityPartsSeparator.ToString())}&lastMax={_lastRangeMax}";
+            var pathBuilder = new StringBuilder();
 
-            var request = new HttpRequestMessage
+            pathBuilder.Append($"{node.Url}/databases/{node.Database}/hilo/next")
+                .Append($"?tag={Uri.EscapeDataString(_tag)}")
```

Comment:
If you are using StringBuilder, don't user interpolation, write it directly to avoid creating temporaries